gravitee-apim-portal-webui-next/src/components/api-filter/api-filter.component.ts

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2024 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { MatButtonToggle, MatButtonToggleGroup } from '@angular/material/button-toggle';
+
+@Component({
+  selector: 'app-api-filter',
+  standalone: true,
+  imports: [MatButtonToggleGroup, MatButtonToggle],
+  templateUrl: './api-filter.component.html',
+  styleUrl: './api-filter.component.scss',
+})
+export class ApiFilterComponent {
+  @Input() preselectedFilterValue!: string;
+  @Output() selectedFilter = new EventEmitter<string>();
+
+  filterOptions: { label: string; value: string }[] = [
+    {
```

Comment:
The list of categories should be generated from a call to the backend to get a list of all categories.